### DISCUSSION TITLE
.csv vs .txt

### DISCUSSION THREAD

**Discussion by CECSpecialistI** [crystalyragui/MARC2RDA#332] [Q&A] [answered]
Because of differences in how .csv is expressed in the U.S. and Holland, Sita can't push changes to our .csv files in Github without ton of extra work/fussing around with formatting and copying/pasting.

I just tried using some .txt files I put in an [experimental folder](https://github.com/uwlib-cams/MARC2RDA/tree/main/Working%20Documents/TESTING%20TAB%20DELIMITED) to make a test change then push, and the only thing that was different for me was that I needed to open my file from Excel rather than from my Windows file explorer, which is not a big deal to me. 

Would it disrupt anyone's workflow if I changed all the working documents from .csv files to tab-delimited text files?  If no one would be extremely inconvenienced, I think it will be best to switch over to .txt, especially as we seek more collaborators from all over the world. @lake44me @gerontakos @AdamSchiff @JianPLee @junghaelee Would you check out the experimental folder of text files before our meeting next week so we can discuss the change then? 

Thanks, everyone, for bearing with me while we get our workflow figured out!

**Comment by gerontakos** [comment]
That will work fine. I even opened a .txt just to make sure and the import wizard makes it easy (as long as we know the delimiter is TAB). So I'm on board with this. I'll make sure I push all my edits today.

**Reply by laura-ake** [reply]
There did not seem to be a way to directly download a particular file from GitHub (maybe I'm showing ignorance?) to my computer outside of GitHub Desktop or GitHub command line (which I'm not using because, Windows).  I found the file in the Windows directory GitHub Desktop set up, right-clicked and selected "open with" and then could not get Windows to suggest Excel, and couldn't find the path to the Excel executable.  
So, I tried opening Excel and then, went hunting for where the file actually lives in my directory structure....  finally I found that Github Desktop had deposited in inside my personal OneDrive folder...  (Windows 10 has ways).  
So then I opened it in Excel and did the usual moves:  Select My File Has Delimiters and My file has headers; specify the delimiter is "Tab", change Text Qualifier to "none", highlight all the columns using click / shift click and select the Column Data Format "Text", and voila.
If someone can suggest another or more streamlined process, I'd be happy but I'm ok for now.  
You might want to write this up for Windows/Excel users.
I wonder if making the file extension ".tab" instead of ".txt" would be useful?  Windows could then associate that extension with Excel ....?

**Reply by CECSpecialistI** [reply]
[Here's what I have as far as a workaround for Windows users](https://github.com/uwlib-cams/MARC2RDA/tree/main/Instructions#default-program-for-text-files). It changes your default program for opening .txt files from File Explorer to Excel. It affects all .txt files, though, so you'd have to right-click and select a different program for other things or change it around all the time.

**Comment by CECSpecialistI** [comment]
Hooray! Thanks Theo

**Comment by CECSpecialistI** [comment, marked as answer]
@gerontakos @laura-ake @lake44me @AdamSchiff @junghaelee @JianPLee @mcm104 I am going to change the files over from .csv to .txt in the next hour or so. If you haven't pushed your .csv changes yet, now is a good time to do that. I figured out how to get the .txt files to automatically open in Excel in Windows from the command line and will update instructions as well.

**Comment by JianPLee** [comment]
Another way for me to open a text file is to drag the file from File Explorer and drop it in a new Excel workbook. It opens instantly without the many clicks.

**Reply by gerontakos** [reply]
Similar to Jian's result, if anyone loves Powershell, you can just enter, from within the directory 4XX.txt is in, the following command:
start excel 4XX.txt
Easy!

**Reply by gerontakos** [reply]
The two methods of opening a file recommended on February 23 by Jian and by Theo ARE NOT VALID WAYS TO OPEN OUR SPREADSHEETS. They can cause numbers to change (see issue #352). Use the method described above by Laura, also on Feb 23, where we first open Excel then complete opening the file using the Text Import Wizard.

**Reply by CECSpecialistI** [reply]
The method I recommended [here](https://github.com/uwlib-cams/MARC2RDA/discussions/332#discussioncomment-2239320) is also not valid, for the same reasons. Removed from instructions spreadsheet on 2022-04-27

**Comment by AdamSchiff** [comment]
We've been looking at transforms for classification and subjects and for $0/$1 in classification and subject fields, but should we also look at examples found in other fields such as 3XX?  How will they be transformed?  Are we going to be minting URIs for these entities too?

Examples:

385 ## Gardeners ǂ2 lcdgt
385 ## Oregonians ǂ2 lcdgt
385 ## Washingtonians (Washington State) ǂ2 lcdgt

386 ## Americans ǂ2 lcdgt
386 ## Oregonians ǂ2 lcdgt
386 ##  Women ǂ2 lcdgt
386 ## ǂ3 Text ǂa Indigenous people of America ǂ2 lcdgt

380 ## Comics (Graphic works) ǂ2 lcgft
385 ## LGBTQ people ǂa Teenagers ǂ2 lcdgt

370 ## ǂi Setting: ǂf Oregon ǂ2 naf ǂ4 stg
370 ## ǂi Setting: ǂf Portland (Or.) ǂ2 naf ǂ4 stg
386 ## Men ǂ2 lcdgt
386 ## Oregonians ǂ2 lcdgt

386 ## Americans ǂa New Yorkers (New York State) ǂa Oregonians ǂa Washingtonians (Washington State) ǂa Seattleites ǂa Women ǂ2 lcdgt

386 ## ǂi Author: ǂa Artists ǂa Art teachers ǂa University and college faculty members ǂa Americans ǂa Oregonians ǂa Eugeneans ǂa Men ǂ2 lcdgt

385 ## Children ǂ2 lcdgt
385 ## Preschool children ǂ2 lcdgt
385 ## Kindergarteners ǂ2 lcdgt
385 ## School children ǂ2 lcdgt
386 ## ǂi Author: ǂa Oregonians ǂ2 lcdgt ǂ4 aut
386 ## ǂi Author: ǂa Women ǂ2 lcdgt ǂ4 aut
386 ## ǂi Illustrator: ǂa Washingtonians (Washington State) ǂ2 lcdgt ǂ4 ill
386 ## ǂi Illustrator: ǂa Men ǂ2 lcdgt ǂ4 ill

370 ## ǂg Northwest, Pacific ǂ2 lcsh
386 ## Pacific Northwesterners ǂ2 lcdgt
386 ## Americans ǂa Canadians ǂa Alaskans ǂa Oregonians ǂa Washingtonians (Washington State) ǂa Montanans ǂa Idahoans ǂa British Columbians ǂ2 lcdgt
388 1# Twentieth century ǂ2 lcsh
388 1# Twenty-first century ǂ2 lcsh

385 ## Children ǂ2 lcdgt
386 ## English ǂ2 lcdgt
386 ## Britons ǂ2 lcdgt
386 ## Women ǂ2 lcdgt
388 1# Victorian era ǂ2 wikidata ǂ1 http://www.wikidata.org/entity/Q182688

385 ## Preteens ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060394
386 ## French ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060737

257 ## United States ǂ2 naf ǂ0 http://id.loc.gov/authorities/names/n78095330 ǂ1 http://id.loc.gov/rwo/agents/n78095330
257 ## Great Britain ǂ2 naf ǂ0 http://id.loc.gov/authorities/names/n79023147 ǂ1 http://id.loc.gov/rwo/agents/n79023147
336 ## two-dimensional moving image ǂb tdi ǂ2 rdacontent ǂ0 http://id.loc.gov/vocabulary/contentTypes/tdi
337 ## video ǂb v ǂ2 rdamedia ǂ0 http://id.loc.gov/vocabulary/mediaTypes/v
338 ## videodisc ǂb vd ǂ2 rdacarrier ǂ0 http://id.loc.gov/vocabulary/carriers/vd
340 ## ǂb 4 3/4 in.
344 ## digital ǂ2 rdatr ǂ1 http://rdaregistry.info/termList/typeRec/1002
344 ## ǂb optical ǂ2 rdarm ǂ1 http://rdaregistry.info/termList/recMedium/1003
344 ## ǂg surround ǂ2 rdacpc ǂ1 http://rdaregistry.info/termList/configPlayback/1004
344 ## ǂh Dolby Digital 5.1
346 ## ǂb NTSC ǂ2 rdabs ǂ0 http://rdaregistry.info/termList/broadcastStand/1002
347 ## video file ǂ2 rdaft ǂ1 http://rdaregistry.info/termList/fileType/1006
347 ## ǂb DVD video
347 ## ǂe region 1 ǂ2 rdare ǂ1 http://rdaregistry.info/termList/RDARegionalEncoding/1002
380 ## Motion pictures ǂ2 lcgft ǂ0 http://id.loc.gov/authorities/genreForms/gf2011026406
386 ## ǂi Film director: ǂa Mexicans ǂ2 lcdgt ǂ4 http://id.loc.gov/vocabulary/relators/fmd ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060329
386 ## ǂi Film director: ǂa Men ǂ2 lcdgt ǂ4 http://id.loc.gov/vocabulary/relators/fmd ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060359

380 ## LGBTQ+ graphic novels ǂ2 homoit
380 ## LGBTQ+ comics ǂ2 homoit
380 ## Gay comics ǂ2 lcgft
380 ## Queer comics ǂ2 lcgft
380 ## Graphic novels ǂ2 lcgft
385 ## LGBTQ+ people ǂ2 homoit
385 ## Queer people ǂ2 homoit

386 ## Americans ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060001
386 ## Transgender people ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060006
386 ## Queer people ǂ2 homoit ǂ0 https://homosaurus.org/v3/homoit0001195
386 ## Non-binary people ǂ2 homoit ǂ0 https://homosaurus.org/v3/homoit0001048

370 ## ǂi Setting: ǂf Spokane Indian Reservation (Wash.) ǂ2 lcsh ǂ0 http://id.loc.gov/authorities/subjects/sh90003893
370 ## ǂi Setting: ǂf Seattle (Wash.) ǂ2 naf ǂ0 http://id.loc.gov/authorities/names/n79041965 ǂ1 http://id.loc.gov/rwo/agents/n79041965
386 ## Seattleites ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2022060019 ǂ1 http://www.wikidata.org/entity/Q55779668
386 ## Washingtonians (Washington State) ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060124 ǂ1 http://www.wikidata.org/entity/Q55590658
386 ## Americans ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060001 ǂ1 http://www.wikidata.org/entity/Q846570
386 ## Men ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060359 ǂ1 http://www.wikidata.org/entity/Q8441
386 ## Coeur d'Alene (North American people) ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2016060401 ǂ1 http://www.wikidata.org/entity/Q491755
386 ## Coeur d'Alene Indians ǂ2 lcsh ǂ0 http://id.loc.gov/authorities/subjects/sh85123215  ǂ1 http://www.wikidata.org/entity/Q491755
386 ## Spokane (North American people) ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2016060400  ǂ1 http://www.wikidata.org/entity/Q491709
386 ## Spokane Indians ǂ2 lcsh ǂ0 http://id.loc.gov/authorities/subjects/sh85126844 ǂ1 http://www.wikidata.org/entity/Q491709

386 ## ǂ3 1st work ǂi Composer: ǂa Americans ǂa Women ǂa Lesbians ǂ2 lcdgt
386 ## ǂ3 2nd work ǂi Composer: ǂa Russians ǂa Men ǂ2 lcdgt

370 ## ǂi Place of capture: ǂf Kentish Town (London, England) ǂ2 naf ǂ4 http://rdaregistry.info/Elements/u/P60556 ǂ0 http://id.loc.gov/authorities/names/nb2005005089 ǂ1 http://id.loc.gov/rwo/agents/nb2005005089
382 01 piano ǂn 1 ǂs 1 ǂ2 lcmpt ǂ0 http://id.loc.gov/authorities/performanceMediums/mp2013015550
386 ## ǂi Performer: ǂa Russians ǂ2 lcdgt ǂ4 http://id.loc.gov/vocabulary/relators/prf ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060814
386 ## ǂi Performer: ǂa Men ǂ2 lcdgt ǂ4 http://id.loc.gov/vocabulary/relators/prf ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060359

386 ## ǂ3 Witches' dance ; Burlesque ǂi Composer: ǂa Italians ǂ2 lcdgt
386 ## ǂ3 Romance andaluso ; Zapateado ; Spanish dance ; Malagueña ǂi Composer: ǂa Spaniards ǂ2 lcdgt
386 ## ǂ3 Love's sorrow ǂi Composer: ǂa French ǂ2 lcdgt
386 ## ǂ3 Dance russe from Petrouchka suite ǂi Composer: ǂa Russians ǂ2 lcdgt
386 ## ǂ3 Sicilienne ǂi Composer: ǂa Americans ǂ2 lcdgt
386 ## ǂ3 Appassionata ǂi Composer: ǂa Czechs ǂ2 lcdgt
386 ## ǂ3 Waltz in A major, Lullaby ǂi Composer: ǂa Germans ǂ2 lcdgt
386 ## ǂi Composer: ǂa Men ǂ2 lcdgt

382 0# ǂ3 Cello concerto ; Schelomo ; Cello concerto no. 2 ; Variations on a rococo theme ; Don Quixote ǂb cello ǂn 1 ǂa orchestra ǂe 1 ǂr 1 ǂt 1 ǂ2 lcmpt
382 0# ǂ3 Nocturne ; Pezzo capriccioso ; Andante cantabile ǂa orchestra ǂe 1 ǂt 1 ǂ2 lcmpt ǂ0 http://id.loc.gov/authorities/performanceMediums/mp2013015516
386 ## ǂ3 Cello concerto ǂi Composer: ǂa English ǂa Britons ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060386 ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060373
386 ## ǂ3 Schelomo ǂi Composer: ǂa Swiss ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060375
386 ## ǂ3 Cello concerto no. 2 ; Variations on a rococo theme ; Nocturne ; Pezzo capriccioso ; Andante cantabile ǂi Composer: ǂa Russians ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060814
386 ## ǂ3 Don Quixote ǂi Composer: ǂa Germans ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060340
386 ## ǂi Composer: ǂa Men ǂ2 lcdgt ǂ0 http://id.loc.gov/authorities/demographicTerms/dg2015060359


Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Reply by GordonDunsire** [reply]
In the context of subject classifications and headings, the primary purpose of minting a concept IRI is to retain the data provenance for the source VES. A secondary purpose is to account for synthesized concepts that do not have an IRI in the source VES.

This is not necessary if the VES IRI is recorded in $0/$1; as RDA Toolkit points out, a de-referenceable IRI carries its own provenance. If the VES IRI is not recorded, the minted transform IRI can be mapped to it during subsequent processing.

In the case of 38x, I think the model is similar to subfield $v in subject headings as a category of work. The fields essentially contain information about categories of agents who create works embodied by the manifestation being described. The basic model is:

<work> rdawo:P10065 <agent> . // has creator agent of work; there is no data about the type of agent. The property can be refined by the relator information
<agent> rdaao:P50490 <concept> . // has category of agent

The issues are: which work? which agent?

The $0/$1 issue is: is it safe to assume that only PCC guidance is used in treating $1 as $0? If we say yes to FAST subject headings, we should say yes to demographic terms, etc. But only after resolving which work and which agent.
